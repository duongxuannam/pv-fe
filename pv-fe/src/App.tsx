import { useState } from "react"

const data = [
  {
    id: 1,
    name: "Test 1",
  },
  {
    id: 2,
    name: "Test 2",
  },
  {
    id: 3,
    name: "Test 3",
  },
  {
    id: 4,
    name: "Test 4",
  },
]
const Div = ({ children }: any) => {
  return (
    <div
      style={{
        width: 400,
      }}
    >
      {children}
    </div>
  )
}
function App() {
  const [idModal, setIdModal] = useState()
  const handleOnclick = (id: any) => {
    setIdModal(id)
  }
  const itemModal = data.find((e: any) => e.id === idModal)
  console.log("id", idModal, itemModal)

  return (
    <div
      className="min-w-[375px]"
      style={{
        position: "relative",
      }}
    >
      {/* <User />
      <Devices /> */}
      {/* modal here */}
      {itemModal && (
        <div
          style={{
            position: "absolute",
            height: "100vh",
            width: "100%",
            backgroundColor: "black",
            opacity: 0.4,
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
          }}
        >
          <div
            style={{
              width: 300,
              aspectRatio: 1,
              backgroundColor: "white",
            }}
          >
            <div
              onClick={() => setIdModal(undefined)}
              style={{ display: "flex", justifyContent: "flex-end" }}
            >
              Close
            </div>
            <div
              style={{
                display: "flex",
                gap: 80,
              }}
            >
              <Div>
                <span>Id: {itemModal.id}</span>
              </Div>
              <Div>
                <span>Name: {itemModal.name}</span>
              </Div>
              <Div>
                <span>Name: {itemModal.name}</span>
              </Div>
              <Div>
                <span>Name: {itemModal.name}</span>
              </Div>
              <Div>
                <span onClick={() => handleOnclick(itemModal.id)}>Onclick</span>
              </Div>
            </div>
          </div>
        </div>
      )}
      demo
      <div
        style={{
          display: "flex",
          gap: 80,
        }}
      >
        <Div>
          <span>Id</span>
        </Div>
        <Div>
          <span>Name</span>
        </Div>
        <Div>
          <span>Content</span>
        </Div>
        <Div>
          <span>Content</span>
        </Div>
        <Div>
          <span>Content</span>
        </Div>
      </div>
      {data.map((item, index) => {
        return (
          <div
            style={{
              display: "flex",
              gap: 80,
            }}
          >
            <Div>
              <span>Id: {item.id}</span>
            </Div>
            <Div>
              <span>Name: {item.name}</span>
            </Div>
            <Div>
              <span>Name: {item.name}</span>
            </Div>
            <Div>
              <span>Name: {item.name}</span>
            </Div>
            <Div>
              <span onClick={() => handleOnclick(item.id)}>Onclick</span>
            </Div>
          </div>
        )
      })}
    </div>
  )
}

export default App
