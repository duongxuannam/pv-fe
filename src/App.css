

.App-logo {
  height: 40vmin;
  pointer-events: none;
}



.App-header {
  background-color: red;
  min-height: 80px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  font-size: calc(10px + 2vmin);
}

