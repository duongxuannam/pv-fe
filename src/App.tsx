import { User } from "./features/user/User"
import Devices from "./features/devices/Devices"

function App() {
  return (
    <div
      className="min-w-[375px]"
      style={{
        position: "relative",
      }}
    >
      <User />
      <Devices />
    </div>
  )
}

export default App
