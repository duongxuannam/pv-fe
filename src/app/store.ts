import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit"
import { setupListeners } from "@reduxjs/toolkit/query"
import userReducer from "../features/user/userSlice"
import { devicesApi } from "../features/devices/devicesAPI"

export const store = configureStore({
  reducer: {
    user: userReducer,
    [devicesApi.reducerPath]: devicesApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(devicesApi.middleware),
})

setupListeners(store.dispatch)

export type AppStore = typeof store
export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
