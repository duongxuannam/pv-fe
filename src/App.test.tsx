import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "./app/store"
import App from "./App"

test("Check snapshot", () => {
  const view = render(
    <Provider store={store}>
      <App />
    </Provider>,
  )
  expect(view).toMatchSnapshot()
})
