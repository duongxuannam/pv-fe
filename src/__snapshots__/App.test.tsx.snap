// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="min-w-[375px]"
      >
        <header
          class="h-16 bg-sky-500  flex justify-end items-center"
        >
          <h1
            class="text-lg font-bold underline text-zinc-950"
            data-testid="name"
          />
          <img
            alt="Rounded avatar"
            class="w-10 h-10 rounded-full mx-5"
            src="https://scontent.fsgn2-7.fna.fbcdn.net/v/t1.18169-1/11193438_402423169882782_2597021278587966343_n.jpg?stp=c0.2.40.40a_cp0_dst-jpg_p40x40&_nc_cat=108&ccb=1-7&_nc_sid=7206a8&_nc_ohc=iCEUNNQblrkAX91dsZP&_nc_ht=scontent.fsgn2-7.fna&oh=00_AfClD5sQ-_x_MYzw8cqGVumh1oerFns1BnnwjxMTq8Bhtg&oe=64EC1BA8"
          />
        </header>
        <div
          class="flex justify-end mt-10 mr-10"
        >
          <button
            class=" text-end bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
          >
            Refresh
          </button>
        </div>
        <div
          class="m-10"
        >
          <div
            class="grid gap-5 2 grid-cols-1 md:grid-cols-2 lg:grid-cols-4 xl:grid-cols-4"
          >
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
            <div
              class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
            >
              <div
                class="h-60 w-max bg-gray-300 dark:bg-gray-600"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="min-w-[375px]"
    >
      <header
        class="h-16 bg-sky-500  flex justify-end items-center"
      >
        <h1
          class="text-lg font-bold underline text-zinc-950"
          data-testid="name"
        />
        <img
          alt="Rounded avatar"
          class="w-10 h-10 rounded-full mx-5"
          src="https://scontent.fsgn2-7.fna.fbcdn.net/v/t1.18169-1/11193438_402423169882782_2597021278587966343_n.jpg?stp=c0.2.40.40a_cp0_dst-jpg_p40x40&_nc_cat=108&ccb=1-7&_nc_sid=7206a8&_nc_ohc=iCEUNNQblrkAX91dsZP&_nc_ht=scontent.fsgn2-7.fna&oh=00_AfClD5sQ-_x_MYzw8cqGVumh1oerFns1BnnwjxMTq8Bhtg&oe=64EC1BA8"
        />
      </header>
      <div
        class="flex justify-end mt-10 mr-10"
      >
        <button
          class=" text-end bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
        >
          Refresh
        </button>
      </div>
      <div
        class="m-10"
      >
        <div
          class="grid gap-5 2 grid-cols-1 md:grid-cols-2 lg:grid-cols-4 xl:grid-cols-4"
        >
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
          <div
            class="flex w-full max-w-md mx-auto overflow-hidden rounded-lg shadow-lg bg-gray-300 animate-pulse dark:bg-gray-800"
          >
            <div
              class="h-60 w-max bg-gray-300 dark:bg-gray-600"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
